python/py-opt_einsum/Portfile

```diff
@@ -0,0 +1,64 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           python 1.0
+
+name                py-opt_einsum
+version             3.2.1
+revision            0
+
+license             MIT
+maintainers         nomaintainer
+platforms           darwin
+supported_archs     noarch
+
+description         A tensor contraction order optimizer.
+
+long_description    Optimized einsum can significantly reduce the\
+                    overall execution time of einsum-like expressions\
+                    (e.g., np.einsum, dask.array.einsum, pytorch.einsum,\
+                    tensorflow.einsum) by optimizing the expression's\
+                    contraction order and dispatching many operations\
+                    to canonical BLAS, cuBLAS, or other specialized\
+                    routines. Optimized einsum is agnostic to the backend\
+                    and can handle NumPy, Dask, PyTorch, Tensorflow, CuPy,\
+                    Sparse, Theano, JAX, and Autograd arrays as well as\
+                    potentially any library which conforms to a standard API.
+
+homepage            https://github.com/dgasmith/opt_einsum
+
+checksums           rmd160  7a3ab41ecc26559b0f01ecfbc93453e7c84ab3ef \
+                    sha256  83b76a98d18ae6a5cc7a0d88955a7f74881f0e567a0f4c949d24c942753eb998 \
+                    size    72186
+
+supported_archs     noarch
+                    
+python.versions     37 38
+
+if {${name} ne ${subport}} {
+    depends_build-append \
+                    port:py${python.version}-setuptools
+
+    depends_lib-append \
+                    port:py${python.version}-numpy
+
+    depends_test-append \
+                    port:py${python.version}-pytest \
+                    port:py${python.version}-pytest-cov \
+                    port:py${python.version}-pep8
```

Comment:
I think this should be `pytest-pep8`

Reply:
There is no `pytest-pep8`.

Also, this is the only way I was able to get the test to run, and the actual test in upstream is broken. At least this runs the broken test code…

Reply:
there is certainly a `pytest-pep8` on [PyPI](https://pypi.org/project/pytest-pep8/), not in MacPorts indeed. These dependencies are not needed you can remove them and run the tests in trace-mode with `-t` and you'll see it still runs and has the same test failures.

One thing is that upstream apparently does not list `theano` as a test dependency, but than does use it in `tests/test_backends.py`. But okay, I likely wouldn't add `theano` as a test dependency for this and these failure are fine with me. Not good is to add dependencies that are not needed, so please remove them.

Reply:
Removed thanks.